crates/ra_syntax/src/lib.rs

```diff
@@ -160,6 +160,15 @@ impl SourceFile {
     }
 }
 
+impl Parse<ast::Expr> {
+    pub fn parse(s: &str) -> Parse<ast::Expr> {
```

Comment:
This seems very specific for macro expansion, so let's keep the definition in the same crate that uses it.